x/metadata/keeper/mocks_test.go

```diff
@@ -325,14 +332,323 @@ func (a *MockAuthorization) Reset() {
+
+// MockMarkerKeeper is a mocked keeper.MarkerKeeper.
+type MockMarkerKeeper struct {
+	IsMarkerAccountResults map[string]bool
+	IsMarkerAccountCalls   []sdk.AccAddress
+}
+
+func NewMockMarkerKeeper() *MockMarkerKeeper {
+	return &MockMarkerKeeper{
+		IsMarkerAccountResults: make(map[string]bool),
+	}
+}
+
+// WithIsMarkerAccountResults sets up the provided addrs to return true from IsMarkerAccount.
+func (k *MockMarkerKeeper) WithIsMarkerAccountResults(addrs ...sdk.AccAddress) *MockMarkerKeeper {
+	for _, addr := range addrs {
+		k.IsMarkerAccountResults[string(addr)] = true
+	}
+	return k
+}
+
+// ClearResults clears previously recorded calls but leaves the desired results intact.
+func (k *MockMarkerKeeper) ClearResults() {
+	k.IsMarkerAccountCalls = nil
+}
+
+// AssertIsMarkerAccountCalls asserts that calls made to IsMarkerAccount are as expected.
+func (k *MockMarkerKeeper) AssertIsMarkerAccountCalls(t *testing.T, exp []sdk.AccAddress) bool {
+	t.Helper()
+	act := k.IsMarkerAccountCalls
+	if assert.Equal(t, exp, act, "Addrs provided to IsMarkerAccount") {
+		return true
+	}
+	expStrs := addrsCastToStrings(exp)
+	actStrs := addrsCastToStrings(act)
+	assert.Equal(t, expStrs, actStrs, "Addrs (as strings) provided to IsMarkerAccount")
+	return false
+}
+
+func (k *MockMarkerKeeper) GetMarkerByDenom(_ sdk.Context, _ string) (markertypes.MarkerAccountI, error) {
+	panic("MockMarkerKeeper.GetMarkerByDenom not implemented")
+}
+
+func (k *MockMarkerKeeper) IsMarkerAccount(_ sdk.Context, addr sdk.AccAddress) bool {
+	k.IsMarkerAccountCalls = append(k.IsMarkerAccountCalls, addr)
+	return k.IsMarkerAccountResults[string(addr)]
+}
+
+// ensure that the MockBankKeeper implements keeper.BankKeeper.
+var _ keeper.BankKeeper = (*MockBankKeeper)(nil)
+
+// MockBankKeeper is a mocked keeper.BankKeeper.
+type MockBankKeeper struct {
+	BlockedAddrResults map[string]bool
+	MintCoinsResults   []string
+	BurnCoinsResults   []string
+	SendCoinsResults   map[string]string
+	DenomOwnerResults  map[string]DenomOwnerResult
+
+	Calls BankKeeperCalls
+}
+
+// BankKeeperCalls contains records of calls made to the mock bank keeper.
+type BankKeeperCalls struct {
+	BlockedAddr []sdk.AccAddress
+	MintCoins   []*MintBurnCall
+	BurnCoins   []*MintBurnCall
+	SendCoins   []*SendCoinsCall
+	DenomOwner  []string
+}
+
+// NewMockBankKeeper creates a new MockBankKeeper.
+// Usually followed by calls to WithBlockedAddr, WithMintCoinsErrors, WithBurnCoinsErrors,
+// SendCoinsErrors, WithDenomOwnerResult, and/or WithDenomOwnerError.
+func NewMockBankKeeper() *MockBankKeeper {
+	return &MockBankKeeper{
+		BlockedAddrResults: make(map[string]bool),
+		SendCoinsResults:   make(map[string]string),
+		DenomOwnerResults:  make(map[string]DenomOwnerResult),
+	}
+}
+
+// WithBlockedAddr makes the provided addr report as blocked.
+func (k *MockBankKeeper) WithBlockedAddr(addr sdk.AccAddress) *MockBankKeeper {
+	k.BlockedAddrResults[string(addr)] = true
+	return k
```

Comment:
Just like the other similar places you've pointed this out, casting is used for these map keys because it's significantly faster than calling `.String()` (for `AccAddress`es).